src/Interpreters/Context.cpp

```diff
@@ -1973,6 +1973,12 @@ void Context::reloadConfig() const
 
 void Context::shutdown()
 {
+    for (auto & [disk_name, disk] : getDisksMap())
+    {
+        LOG_INFO(shared->log, "Shutdown disk {}", disk_name);
+        disk->shutdown();
```

Comment:
But tables shutdown will be called only after this line in `shared->shutdown()`. Can it lead to unexpected errors in log on server shutdown?

Reply:
`client->DisableRequestProcessing` stops any next retry attempt for ongoing S3 request.
If an S3 request is failed and the code above is executed S3 client immediately returns the last failed S3 request outcome.
If S3 is healthy nothing wrong will be happened and S3 requests are processed in a regular way without errors.
If S3 is unhealthy we will not wait till retries are exhausted and fail request forcibly. 
When table is shutdown it waits for currently running merges or moves. If a table uses S3 and S3 is unhealthy we will just fail merge or move the process faster with the same result as we fail request after all unsuccessful retry attempts. If S3 is a healthy merge or move process will finish in a regular way without errors.

Reply:
Awesome, can you add the same line to the `shutdown` method of DiskS3.